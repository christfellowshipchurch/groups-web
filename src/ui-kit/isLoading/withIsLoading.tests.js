import React from 'react';
import renderer from 'react-test-renderer';

// import Providers from '../../Providers';

import { getIsLoading, withIsLoading } from '.';

describe('the withIsLoading HOC', () => {
  it('should pass the isLoading prop (true) to context for children to consume', () => {
    const ChildComponentGettingContext = getIsLoading(({ isLoading }) =>
      isLoading.toString()
    );

    const ParentComponentPassingConext = withIsLoading(() => (
      <ChildComponentGettingContext />
    ));

    const tree = renderer.create(<ParentComponentPassingConext isLoading />);

    expect(tree).toMatchSnapshot();
  });
});
