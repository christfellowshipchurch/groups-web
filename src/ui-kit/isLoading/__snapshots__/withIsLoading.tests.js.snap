// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the withIsLoading HOC should pass the isLoading prop (true) to context for children to consume 1`] = `"true"`;
